### PR TITLE
cmd/derper: add /generate_204 endpoint

### PR DESCRIPTION
**Pull request by andrew-d**
For captive portal detection.

Signed-off-by: Andrew Dunham <andrew@tailscale.com>

### CODE REVIEW

File: cmd/derper/derper.go [hunk 1]
```diff
@@ -293,6 +297,7 @@ func main() {
 		})
 		if *httpPort > -1 {
 			go func() {
+
```
Comment (bradfitz) [inline cmd/derper/derper.go:300]: unrelated?

File: cmd/derper/derper.go [hunk 2]
```diff
@@ -293,6 +297,7 @@ func main() {
...
 				port80srv := &http.Server{
 					Addr:        net.JoinHostPort(listenHost, fmt.Sprintf("%d", *httpPort)),
 					Handler:     certManager.HTTPHandler(tsweb.Port80Handler{Main: mux}),
```
Comment (bradfitz) [inline cmd/derper/derper.go:303]: won't this still redirect /generate_204 to https?

Comment (andrew-d) [reply cmd/derper/derper.go]: Whoops, forgot to add/push the rest of the commit 🤦

File: cmd/derper/derper.go [hunk 3]
```diff
@@ -293,9 +297,19 @@ func main() {
 		})
 		if *httpPort > -1 {
 			go func() {
+				certHandler := certManager.HTTPHandler(tsweb.Port80Handler{Main: mux})
```
Comment (bradfitz) [inline cmd/derper/derper.go:300]: or

```go
    port80mux := http.NewServeMux()
    port80mux.HandleFunc("/generate_204", serve204NoContent)
    port80mux.Handle("/", certManager.HTTPHandler(tsweb.Port80Handler{Main: mux}))
```

(and similar HandleFunc serve204NoContent above)

your call either way

**Review by bradfitz — APPROVED** [review]
